doc/architectural/nondet-volatile.md

```diff
@@ -0,0 +1,136 @@
+\ingroup module_hidden 
```

Comment:
Restoring this is fine, but I had intentionally removed it for its contents was added to the man page (of goto-instrument).

Reply:
Removed (and links from user guide and architectural guide).